opm/autodiff/NonlinearSolver_impl.hpp

```diff
@@ -260,24 +260,28 @@ namespace Opm
         dxOld = dx;
 
         switch (relaxType()) {
-            case DAMPEN:
+            case DAMPEN: {
                 if (omega == 1.) {
                     return;
                 }
-                for (size_t i = 0; i < dx.size(); ++i) {
+                auto i = dx.size();
+                for (i = 0; i < dx.size(); ++i) {
                     dx[i] *= omega;
                 }
                 return;
-            case SOR:
+            }
+            case SOR: {
                 if (omega == 1.) {
                     return;
                 }
-                for (size_t i = 0; i < dx.size(); ++i) {
+                auto i = dx.size();
+                for (i = 0; i < dx.size(); ++i) {
```

Comment:
I am not too happy about using decltype in non-generic code (where we know the types involved). My preferred solution is (optionally with size_t instead of int):
```
int num = dx.size();
for (int i = 0; i < num; ++i) {
...
```
but since nobody but myself seem to prefer this I don't require it...

I think that decltype is still exotic enough that it should only be used where there are no alternatives. Perhaps I am wrong about this?